### ISSUE TITLE
`Debug` formatting floating point number does not (always) include decimal separator

### ISSUE THREAD

**Issue by sanmai-NL** [rust-lang/rust#47536]
In order to keep optimal equivalence between a `Debug` formatted representation of values and Rust source code, it would be helpful if e.g. `f64` values are formatted including their decimal separator by default. This avoids having to adjust a whole lot of `f64` fields in some larger structure when e.q. writing an equivalence assertion of such a value based on earlier `Debug` output. Currently, the values are formatted as integers when they are representable as integers.

## Meta 

```
rustc 1.25.0-nightly (79a521bb9 2018-01-15)
binary: rustc
commit-hash: 79a521bb9a8ace1a6663578a4c409906adde620d
commit-date: 2018-01-15
host: x86_64-unknown-linux-gnu
release: 1.25.0-nightly
LLVM version: 4.0
```

**Comment by varkor** [comment]
[This seems to be the current behaviour](https://play.rust-lang.org/?gist=9d4f904fd84f0c6c958365be6eea4d34&version=nightly) as of https://github.com/rust-lang/rust/commit/3e98f18280a9204f9476f3a7ffc83782ea81dfdf. Are you sure you weren't using `Display` instead, or an older version of the compiler?

**Comment by dtolnay** [comment]
Thanks! This has been fixed as of #46831 and will be released in Rust 1.24.0.

**Comment by sanmai-NL** [comment]
@varkor: Yeah, I was using an older version of the compiler (different toolchain).
